ding/world_model/base_world_model.py

```diff
@@ -215,10 +215,12 @@ def step(obs, act):
             # terminals = self.termination_fn(next_obs)
             timesteps = {
                 id: BaseEnvTimestep(n, r, d, {})
-                for id, n, r, d in zip(data_id,
-                                       next_obs.cpu().numpy(),
-                                       rewards.cpu().numpy(),
-                                       terminals.cpu().numpy())
+                for id, n, r, d in zip(
+                    data_id,
+                    next_obs.cpu().numpy(),
+                    rewards.unsqueeze(-1).cpu().numpy(),  # ding api
```

Comment:
can we define reward.unsqueeze in model-based env, just like other real envs

Reply:
The short answer is maybe we can't.
`(B, 1)` size is only necessary when `WorldModel` has to interact with `IBuffer`. The data fetched from the buffer is then passed to `default_collate` to squeeze the trailing dimension to be `(B,)`.
However, in some other cases, as in `mbsac`, `WorldModel` interacts with the policy directly and produces batched data (since transitions are now outputs from the neurel nets) so the best output size is `(B, )`, otherwise I need to squeeze here and there in the policy side, which is redundant.

Reply:
okk